agent/config/config_test.go

```diff
@@ -67,6 +67,25 @@ func TestBrokenEC2MetadataEndpoint(t *testing.T) {
 	assert.Zero(t, config.APIEndpoint, "Endpoint env variable not set; endpoint should be blank")
 }
 
+func TestGetRegionWithNoIID(t *testing.T) {
+	defer setTestEnv("AWS_DEFAULT_REGION", "")()
```

Comment:
Can we change this to be setTestRegion() which sets region to us-west-2? Just to make this consistent with other tests.

Reply:
I changed it to use us-east-1 so that it is distinctively different from other tests, which all uses setTestRegion() convenience method that sets the region to use-west-2.

Reply:
But since this is a defer, you are resetting the TestEnv var. Shouldn't it reset TestEnv in the same manner as other tests, regardless of how it is used in this one specific test?

Reply:
it sets the env var immediately and defers the unset call (see [here](https://github.com/aws/amazon-ecs-agent/blob/master/agent/config/config_test.go#L623)).
What value we are using in the test shouldn't matter, as long as we unset it between tests and make sure one test's state is leaked into the subsequent tests.